pkg/client/app.go

```diff
@@ -137,6 +138,14 @@ func ToAppUpdate(ctx context.Context, c Client, name string, opts *AppUpdateOpti
 	if opts.AutoUpgradeInterval != "" {
 		app.Spec.AutoUpgradeInterval = opts.AutoUpgradeInterval
 	}
+	if len(opts.Memory.Workloads) != 0 {
+		app.Spec.Memory.Workloads = opts.Memory.Workloads
+	}
+	if opts.Memory.All != nil {
+		app.Spec.Memory.All = opts.Memory.All
+		// TODO: needed?
+		app.Spec.Memory.Workloads = opts.Memory.Workloads
```

Comment:
That's how I was reading it too, completely forgot that I had a TODO comment and it got committed here. I ended up overhauling how we set memory on the `appSpec` in general so that's definitely worth another look if you've got the time.